BUILD.bazel

```diff
@@ -200,4 +200,23 @@ proto_library(
         "@com_google_protobuf//:duration_proto",
         "@com_google_protobuf//:wrappers_proto",
     ],
-  )
+)
+
+proto_library(
+    name = "meshca_service_proto",
+    srcs = ["grpc/tls/provider/meshca/v1/service.proto"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@com_google_protobuf//:duration_proto",
+    ],
+)
+
+proto_library(
+    name = "meshca_config_proto",
+    srcs = ["grpc/tls/provider/meshca/v1/config.proto"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@com_google_protobuf//:duration_proto",
+        "@envoy_api//envoy/config/core/v3:pkg",
```

Comment:
We should avoid this dependency in the main BUILD. Put the BUILD in grpc/tls/provider/meshca/v1 instead. If we need to, we can use an alias here, but it's probably fine for users to just reference the directory directly.

Reply:
(Avoiding the dependency here will allow users that don't use meshca to avoid the envoy_api dependency in their WORKSPACE file.)

Reply:
Done.
But I don't quite understand how this affects dependencies for people who don't use the meshca protos.

Reply:
Hmmm... Looks like I am mistaken. As long as the target isn't used, the dep doesn't have to be known to Bazel.

It would be a problem if we needed some weird import in the BUILD ("load"), but not like it was. Sorry.

Reply:
Is there a reason we prefer a single top-level BUILD file instead of BUILD files in each directory?  

And would you prefer that I revert to having the meshca build rules in the top-level BUILD file?

Reply:
I don't know why we are using a single mega-BUILD. It would be good to use BUILD-per-folder.